### PR TITLE
Fix iface_options on s390x

### PR DESCRIPTION
**Pull request by smitterl**
1. iface_options.cfg: Use iface_model "virtio" which is available on all platforms.

2. iface_options.py:
 a. host_arch: Use host architecture info to set expected command line option
 b. pci in model_option: vectors argument only valid for PCI

Test run on s390x:
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system virtual_network --vt-only-filter virtual_network.iface_options.iface_source_default,virtual_network.iface_options.iface_driver.driver_txmode_iothread,virtual_network.iface_options.iface_driver.driver_txmode_timer,virtual_network.iface_options.iface_driver.driver_queues_positive.modify_iface,virtual_network.iface_options.iface_driver.driver_vhost.with_vhost_net,virtual_network.iface_options.iface_driver.driver_vhost.without_vhost_net,virtual_network.iface_options.iface_driver.driver_tx_queue_size.start_success.left_boundary_value,virtual_network.iface_options.iface_driver.driver_tx_queue_size.start_success.right_boundary_value,virtual_network.iface_options.iface_driver.driver_tx_queue_size.start_success.middle_value,virtual_network.iface_options.iface_driver.driver_rx_queue_size.start_success.left_boundary_value,virtual_network.iface_options.iface_driver.driver_rx_queue_size.start_success.right_boundary_value,virtual_network.iface_options.iface_driver.driver_rx_queue_size.start_success.middle_value,virtual_network.iface_options.iface_backend.default_test,virtual_network.iface_options.iface_backend.specific_backend,virtual_network.iface_options.iface_type.type_mcast
JOB ID     : 4d5f4b1dcf56537f6b6011ff53f64eab3082503a
JOB LOG    : /root/avocado/job-results/job-2019-12-17T12.36-4d5f4b1/job.log
 (01/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_source_default: PASS (113.23 s)
 (02/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_txmode_iothread: PASS (60.73 s)
 (03/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_txmode_timer: PASS (74.40 s)
 (04/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_queues_positive.modify_iface: PASS (61.72 s)
 (05/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_vhost.with_vhost_net: PASS (72.89 s)
 (06/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_vhost.without_vhost_net: PASS (72.98 s)
 (07/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_tx_queue_size.start_success.left_boundary_value: PASS (73.58 s)
 (08/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_tx_queue_size.start_success.right_boundary_value: PASS (73.81 s)
 (09/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_tx_queue_size.start_success.middle_value: PASS (73.92 s)
 (10/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_rx_queue_size.start_success.left_boundary_value: PASS (73.83 s)
 (11/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_rx_queue_size.start_success.right_boundary_value: PASS (73.77 s)
 (12/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_driver.driver_rx_queue_size.start_success.middle_value: PASS (73.76 s)
 (13/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_type.type_mcast: PASS (386.83 s)
 (14/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_backend.default_test: PASS (64.69 s)
 (15/15) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_backend.specific_backend: PASS (72.91 s)
RESULTS    : PASS 15 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1428.25 s
```

### CODE REVIEW

**Comment by smitterl** [comment]
@dzhengfy

File: libvirt/tests/cfg/virtual_network/iface_options.cfg [hunk 1]
```diff
@@ -173,7 +173,7 @@
                     iface_source = "{'dev':'eno1','mode':'bridge'}"
                 - type_mcast:
                     iface_type = "mcast"
-                    iface_model = "rtl8139"
+                    iface_model = "virtio"
```
Comment (chunfuwen) [inline libvirt/tests/cfg/virtual_network/iface_options.cfg]: this change may leads to failures in some old libvirt version without supporting virtio. Maybe you can enforce iface_model=virtio  from source code when host_arch=s390x

Comment (smitterl) [reply libvirt/tests/cfg/virtual_network/iface_options.cfg]: OK, didn't know. Thanks, I changed it :)

**Comment by smitterl** [comment]
@chunfuwen 
Updated title after minor update. Rerun affected test case only:
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system virtual_network.iface_options.iface_type.type_mcast
JOB ID     : 14df299a8141e4079075f70b3a237be572a9fa12
JOB LOG    : /root/avocado/job-results/job-2019-12-19T13.33-14df299/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virtual_network.iface_options.iface_type.type_mcast: PASS (1095.82 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1099.30 s
```

**Pull request merged**